// any-llm/dist/embeddingModels/VoyageAIModels.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.VoyageAIEmbeddingModels = void 0;
const types_1 = require("../models/types");
const VOYAGE_PLATFORM_LINK = 'https://voyage.ai/';
const VOYAGE_LARGE_2 = {
    modelId: 'voyage-large-2',
    modelName: 'Voyage Large 2',
    provider: types_1.ModelProvider.VoyageAI,
    platformLink: VOYAGE_PLATFORM_LINK,
    maxContextLength: 16000,
    dimensions: 1536,
    maxInputs: 128,
    queryMode: 'query',
    documentMode: 'document',
    truncationAvailable: true,
    encodingFormats: ['float'],
};
const VOYAGE_CODE_2 = {
    modelId: 'voyage-code-2',
    modelName: 'Voyage Code 2',
    provider: types_1.ModelProvider.VoyageAI,
    platformLink: VOYAGE_PLATFORM_LINK,
    maxContextLength: 16000,
    dimensions: 1536,
    maxInputs: 128,
    queryMode: 'query',
    documentMode: 'document',
    truncationAvailable: true,
    encodingFormats: ['float'],
};
const VOYAGE_2 = {
    modelId: 'voyage-2',
    modelName: 'Voyage 2',
    provider: types_1.ModelProvider.VoyageAI,
    platformLink: VOYAGE_PLATFORM_LINK,
    maxContextLength: 4000,
    dimensions: 1024,
    maxInputs: 128,
    queryMode: 'query',
    documentMode: 'document',
    truncationAvailable: true,
    encodingFormats: ['float'],
};
const VOYAGE_LITE_02_INSTRUCT = {
    modelId: 'voyage-lite-02-instruct',
    modelName: 'Voyage Lite 02 Instruct',
    provider: types_1.ModelProvider.VoyageAI,
    platformLink: VOYAGE_PLATFORM_LINK,
    maxContextLength: 4000,
    dimensions: 1024,
    maxInputs: 128,
    queryMode: 'query',
    documentMode: 'document',
    truncationAvailable: true,
    encodingFormats: ['float'],
};
exports.VoyageAIEmbeddingModels = {
    VOYAGE_LARGE_2,
    VOYAGE_CODE_2,
    VOYAGE_2,
    VOYAGE_LITE_02_INSTRUCT,
};
