// any-llm/dist/models/types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChatRoles = exports.validEnviromentKeys = exports.ModelProvider = void 0;
var ModelProvider;
(function (ModelProvider) {
    ModelProvider["OpenAI"] = "OpenAI";
    ModelProvider["Google"] = "Google";
    ModelProvider["Anthropic"] = "Anthropic";
    ModelProvider["Mistral"] = "Mistral";
    ModelProvider["Groq"] = "Groq";
    ModelProvider["Perplexity"] = "Perplexity";
    ModelProvider["Ollama"] = "Ollama";
    ModelProvider["OpenRouter"] = "OpenRouter";
    ModelProvider["Cohere"] = "Cohere";
    ModelProvider["Azure"] = "Azure";
    ModelProvider["VoyageAI"] = "VoyageAI";
})(ModelProvider || (exports.ModelProvider = ModelProvider = {}));
var validEnviromentKeys;
(function (validEnviromentKeys) {
    validEnviromentKeys["OPENAI_API_KEY"] = "OPENAI_API_KEY";
    validEnviromentKeys["ANTHROPIC_API_KEY"] = "ANTHROPIC_API_KEY";
    validEnviromentKeys["GOOGLE_GEMINI_API_KEY"] = "GOOGLE_GEMINI_API_KEY";
    validEnviromentKeys["MISTRAL_API_KEY"] = "MISTRAL_API_KEY";
    validEnviromentKeys["GROQ_API_KEY"] = "GROQ_API_KEY";
    validEnviromentKeys["PERPLEXITY_API_KEY"] = "PERPLEXITY_API_KEY";
    validEnviromentKeys["COHERE_API_KEY"] = "COHERE_API_KEY";
    validEnviromentKeys["AZURE_OPENAI_API_KEY"] = "AZURE_OPENAI_API_KEY";
    validEnviromentKeys["OPENROUTER_API_KEY"] = "OPENROUTER_API_KEY";
    validEnviromentKeys["VOYAGE_API_KEY"] = "VOYAGE_API_KEY";
    validEnviromentKeys["OPENAI_ORGANIZATION_ID"] = "OPENAI_ORGANIZATION_ID";
    validEnviromentKeys["AZURE_OPENAI_ENDPOINT"] = "AZURE_OPENAI_ENDPOINT";
    validEnviromentKeys["AZURE_OPENAI_DEPLOYMENT_ID"] = "AZURE_OPENAI_DEPLOYMENT_ID";
    validEnviromentKeys["AZURE_EMBEDDINGS_NAME"] = "AZURE_EMBEDDINGS_NAME";
    validEnviromentKeys["AZURE_OPENAI_35_TURBO_ID"] = "AZURE_OPENAI_35_TURBO_ID";
    validEnviromentKeys["AZURE_OPENAI_4_ID"] = "AZURE_OPENAI_4_ID";
    validEnviromentKeys["AZURE_OPENAI_4_TURBO_ID"] = "AZURE_OPENAI_4_TURBO_ID";
    validEnviromentKeys["AZURE_OPENAI_4_TURBO_VISION_ID"] = "AZURE_OPENAI_VISION_ID";
})(validEnviromentKeys || (exports.validEnviromentKeys = validEnviromentKeys = {}));
var ChatRoles;
(function (ChatRoles) {
    ChatRoles["User"] = "user";
    ChatRoles["System"] = "system";
    ChatRoles["Assistant"] = "assistant";
})(ChatRoles || (exports.ChatRoles = ChatRoles = {}));
