// any-llm/dist/chatModels/CohereChatModels.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CohereChatModels = void 0;
const types_1 = require("../models/types");
const COHERE_PLATFORM_LINK = 'https://docs.cohere.com/';
// Cohere Models (UPDATED 3/1624) -----------------------------
const COMMAND_LIGHT = {
    modelId: 'command-light',
    modelName: 'Command Light',
    provider: types_1.ModelProvider.Cohere,
    platformLink: COHERE_PLATFORM_LINK,
    imageInput: false,
    minTemperature: 0.0,
    maxTemperature: 1.0,
    maxTokenOutputLength: 4000,
    maxContextLength: 4000,
};
const COMMAND_LIGHT_NIGHTLY = {
    modelId: 'command-light-nightly',
    modelName: 'Command Light Nightly',
    provider: types_1.ModelProvider.Cohere,
    platformLink: COHERE_PLATFORM_LINK,
    imageInput: false,
    minTemperature: 0.0,
    maxTemperature: 1.0,
    maxTokenOutputLength: 8000,
    maxContextLength: 8000,
};
const COMMAND = {
    modelId: 'command',
    modelName: 'Command',
    provider: types_1.ModelProvider.Cohere,
    platformLink: COHERE_PLATFORM_LINK,
    imageInput: false,
    minTemperature: 0.0,
    maxTemperature: 1.0,
    maxTokenOutputLength: 4000,
    maxContextLength: 4000,
};
const COMMAND_NIGHTLY = {
    modelId: 'command-nightly',
    modelName: 'Command Nightly',
    provider: types_1.ModelProvider.Cohere,
    platformLink: COHERE_PLATFORM_LINK,
    imageInput: false,
    minTemperature: 0.0,
    maxTemperature: 1.0,
    maxTokenOutputLength: 8000,
    maxContextLength: 8000,
};
const COMMAND_R = {
    modelId: 'command-r',
    modelName: 'Command-R',
    provider: types_1.ModelProvider.Cohere,
    platformLink: COHERE_PLATFORM_LINK,
    imageInput: false,
    minTemperature: 0.0,
    maxTemperature: 1.0,
    maxTokenOutputLength: 4000,
    maxContextLength: 128000,
};
exports.CohereChatModels = {
    COMMAND_LIGHT,
    COMMAND_LIGHT_NIGHTLY,
    COMMAND,
    COMMAND_NIGHTLY,
    COMMAND_R,
};
