// any-llm/dist/chatModels/GroqChatModels.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GroqChatModels = void 0;
const types_1 = require("../models/types");
const GROQ_PLATFORM_LINK = 'https://groq.com/';
const LLaMA2_70B = {
    modelId: 'llama2-70b-4096',
    modelName: 'LLaMA2 70b Chat',
    provider: types_1.ModelProvider.Groq,
    platformLink: GROQ_PLATFORM_LINK,
    imageInput: false,
    minTemperature: 0.0,
    maxTemperature: 1.0,
    maxTokenOutputLength: 4096,
    maxContextLength: 4096,
};
const MIXTRAL_8X7B = {
    modelId: 'mixtral-8x7b-32768',
    modelName: 'Mixtral 8x7b Instruct v0.1',
    provider: types_1.ModelProvider.Groq,
    platformLink: GROQ_PLATFORM_LINK,
    imageInput: false,
    minTemperature: 0.0,
    maxTemperature: 1.0,
    maxTokenOutputLength: 4096,
    maxContextLength: 32768,
};
const GEMMA_7B = {
    modelId: 'gemma-7b-it',
    modelName: 'Gemma 7b',
    provider: types_1.ModelProvider.Groq,
    platformLink: GROQ_PLATFORM_LINK,
    imageInput: false,
    minTemperature: 0.0,
    maxTemperature: 1.0,
    maxTokenOutputLength: 8192,
    maxContextLength: 8192,
};
exports.GroqChatModels = {
    MIXTRAL_8X7B,
    LLaMA2_70B,
    GEMMA_7B,
};
